packages/colorpickers/src/elements/ColorSwatch/index.tsx

```diff
@@ -0,0 +1,107 @@
+ */
+
+import React, { forwardRef, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { ThemeContext } from 'styled-components';
+import { Tooltip } from '@zendeskgarden/react-tooltips';
+import { useGrid } from '@zendeskgarden/container-grid';
+import { useId } from '@zendeskgarden/container-utilities';
+import { StyledSwatchButton, StyledCheckIcon, StyledCell, StyledColorSwatch } from '../../styled';
+
+export interface ILabeledColor {
+  value: string;
+  label: string;
+}
+
+export interface IColorSwatchProps {
+  /** Sets the two-dimension array of labeled HEX and RGB/A string colors */
+  colors: ILabeledColor[][];
+  /** Sets the focused row index in a controlled color swatch */
+  rowIndex?: number;
+  /** Sets the focused column index in a controlled color swatch */
+  colIndex?: number;
+  /** Sets the selected row index in a controlled color swatch */
+  selectedRowIndex?: number;
+  /** Sets the selected column index in a controlled color swatch */
+  selectedColIndex?: number;
+  /** Sets the default focused row index in a uncontrolled color swatch */
+  defaultRowIndex?: number;
+  /** Sets the default focused column index in a uncontrolled color swatch */
+  defaultColIndex?: number;
+  /** Sets the default selected row index in a uncontrolled color swatch */
+  defaultSelectedRowIndex?: number;
+  /** Sets the default selected column index in a uncontrolled color swatch */
+  defaultSelectedColIndex?: number;
+  /** Handles color swatch changes */
+  onChange?: (rowIndex: number, colIndex: number) => void;
+  /** Handles color swatch select event */
+  onSelect?: (rowIndex: number, colIndex: number) => void;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLTableElement>
+ */
+export const ColorSwatch = forwardRef<HTMLTableElement, IColorSwatchProps>(
+  ({ colors, ...props }, ref) => {
+    const { rtl } = useContext(ThemeContext);
+    const { getGridCellProps } = useGrid({
+      rtl,
+      matrix: colors,
+      selection: true,
+      wrap: true,
+      idPrefix: useId(),
+      ...props
+    });
+
+    return (
+      <StyledColorSwatch ref={ref}>
+        <tbody>
+          {colors.map((row: ILabeledColor[], rowIdx: number) => (
+            <tr key={row[0].value}>
+              {row.map((color: ILabeledColor, colIdx: number) => {
+                const { label, value } = color;
+                const gridCellProps = getGridCellProps({
+                  colIdx,
+                  rowIdx,
+                  type: 'button',
+                  role: undefined
+                });
+
+                return (
+                  <StyledCell key={value}>
+                    <Tooltip content={label}>
+                      <StyledSwatchButton backgroundColor={value} {...gridCellProps}>
+                        <StyledCheckIcon color={value} selected={gridCellProps['aria-selected']} />
+                      </StyledSwatchButton>
+                    </Tooltip>
+                  </StyledCell>
```

Comment:
The a11y is improved here. But the VO and interaction pattern is still a bit off.
- buttons should be able to toggle on and off
- button state is not reported to VO – I think these will either need to be toggle (`aria-pressed`) buttons or coerced to be `role=checkbox` with `aria-checked` controls
- there appears to be a problem (previously mentioned by @mtomcal) where the grid hook is allowing direct cursor navigation (on right keypress) to the tooltip body – this should not be happening.

Let's schedule some time to pair on this and work out the expected a11y semantics.